rspace/src/test/scala/coop/rchain/rspace/test/ArbitraryInstances.scala

```diff
@@ -84,6 +84,24 @@ object ArbitraryInstances {
         .map(_.toMap))
   }
 
+  implicit val arbitraryWaitingContinuation
+    : Arbitrary[WaitingContinuation[Pattern, StringsCaptor]] = {
+    Arbitrary(for {
+      chans   <- Gen.containerOf[List, String](Arbitrary.arbitrary[String])
+      pats    <- Gen.containerOf[List, Pattern](Arbitrary.arbitrary[Pattern])
```

Comment:
Hmm.  `consume` has the invariant that the `List[P]` (patterns) needs to be the same length as `List[C]` (channels), so while this works with the cases at hand, it doesn't work in the general case.  Check out below how I'm giving them the same length.